src/python/gudhi/representations/metrics.py

```diff
@@ -116,33 +192,54 @@ def transform(self, X):
         Returns:
             numpy array of shape (number of diagrams in **diagrams**) x (number of diagrams in X): matrix of pairwise bottleneck distances.
         """
-        num_diag1 = len(X)
+        if not USE_GUDHI:
+            print("Gudhi built without CGAL: returning a null matrix")
```

Comment:
(#200 is trying to move those messages to stderr)
I wonder if raising an exception may be preferable. What are the use cases for returning a zero matrix? Is it for a case where you compute distance matrices for several distances and try to learn something from all of them together, and then having one as 0 just means any learning algorithm will discard it as not informative?

Reply:
I totally agree with you. This 0 matrix has no motivation, I also think an exception is better. Just modified the code.